### ISSUE TITLE
Linux

### ISSUE THREAD

**Issue by drq123456** [anliyuan/Ultralight-Digital-Human#33]
"Can it run under Linux ARM64?"

**Comment by anliyuan** [comment]
As long as the inference framework allows

**Comment by drq123456** [comment]
> As long as the inference framework allows
"My framework does not have NVIDIA GPUs, so I can't install CUDA. Without CUDA support, can this digital human run successfully?"

**Comment by anliyuan** [comment]
absolutely yes, I ran it on mobile device.

**Comment by drq123456** [comment]
> absolutely yes, I ran it on mobile device.

"Does that mean I can run it without downloading CUDA and just make some simple modifications to the code?"

**Comment by anliyuan** [comment]
yes, but if you run it on cpu, you cant use hubert encoder, use wenet with 20fps video

**Comment by drq123456** [comment]
> yes, but if you run it on cpu, you cant use hubert encoder, use wenet with 20fps video

Thank you very much！

**Comment by do-one-thing-to-well** [comment]
en，，，sorry to interrupt，，，How to run this on mobile device?

**Comment by drq123456** [comment]
> yes, but if you run it on cpu, you cant use hubert encoder, use wenet with 20fps video

"I followed your steps, but I'm still getting a CUDA error. Is the problem with encoder.onnx?"
![Uploading 微信图片_20241030153157.png…]()

**Comment by anliyuan** [comment]
You want to train without cuda? You neet to put all data and model on cpu then.

**Comment by drq123456** [comment]
> You want to train without cuda? You neet to put all data and model on cpu then.

OKOK thanks

**Comment by drq123456** [comment]
> You want to train without cuda? You neet to put all data and model on cpu then.

conda install pytorch==1.13.1 torchvision==0.14.1 torchaudio==0.13.1 pytorch-cuda=11.7 -c pytorch -c nvidia    
The versions don't need to be exactly the same, right? Mine are slightly lower than these.

**Comment by anliyuan** [comment]
> > You want to train without cuda? You neet to put all data and model on cpu then.
> 
> conda install pytorch==1.13.1 torchvision==0.14.1 torchaudio==0.13.1 pytorch-cuda=11.7 -c pytorch -c nvidia The versions don't need to be exactly the same, right? Mine are slightly lower than these.

Right.
